qemu/tests/nvdimm_mapsync.py

```diff
@@ -0,0 +1,53 @@
+import re
+import pathlib
+
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Run nvdimm cases:
+    1) Mount nvdimm device on host
+    2) Create a file in the mount point
+    3) Boot guest with nvdimm backed by the file
+    4) Check flag 'sf' is present in qemu smaps
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    dev_path = params["dev_path"]
+    p = pathlib.Path(dev_path)
+    if not p.is_block_device():
+        test.error("There is no nvdimm device in host, please add kernel param"
+                   "'memmap' to emulate one")
+
+    format_cmd = params["format_command"]
+    mount_cmd = params["mount_command"]
+    truncate_cmd = params["truncate_command"]
+    check_cmd = params["check_command"]
+    clean_cmd = params["clean_command"]
+    try:
+        process.run(format_cmd)
+        process.run(mount_cmd, shell=True)
+        process.run(truncate_cmd)
+    except Exception as e:
+        test.error(e)
+    else:
+        params["start_vm"] = "yes"
+        env_process.preprocess_vm(test, params, env, params['main_vm'])
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        vm.wait_for_login()
+        vm_pid = vm.get_pid()
+
+        output = process.system_output(check_cmd % vm_pid).decode()
+        if 'sf' not in re.findall(r'VmFlags:.*', output)[0].split():
+            test.fail("Flag 'sf' is not present in smaps file")
+    finally:
+        vm.destroy()
```

Comment:
If ` test.error(e)`, no `vm`

Reply:
And, `clean_cmd` hit error too, if the pre steps `mount` failed

Reply:
For the first issue, we can use a second 'try' instead of the 'else'. As to the second one, what would you suggest? If any step in L35~L37 fails, the clean_cmd would fail.

Reply:
For the second issue, i think a clean up function is needed, check status first before `rm/umount`.

Reply:
Updated, thx.

Reply:
The first issue still exist, when `test.error(e)`, no 'vm' in finally

Reply:
Sorry, forgot to update that one.